File: Java/par_impar.java
import java.util.Locale;
import java.util.Scanner;

public class par_impar {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        Locale.setDefault(Locale.US);

        System.out.print("Quantos números você vai digitar? ");
        int casos = sc.nextInt();

        for (int i = 0; i < casos; i++) {
            System.out.print("Digite um número: ");
            int x = sc.nextInt();
            if (x == 0) {
                System.out.printf("Nulo %n");
            }
            else if (x % 2 != 0) {
                System.out.print("Ímpar ");
                if (x > 0) {
                    System.out.printf("Positivo %n");
                }
                else {
                    System.out.printf("Negativo %n");
                }
            }
            else {
                System.out.print("Par ");
                if (x > 0) {
                    System.out.printf("Positivo %n");
                }
                else {
                    System.out.printf("Negativo %n");
                }
            }
        }

        sc.close();

    }
}

File: Python/par_impar.py
casos = int(input("Quantos números você vai digitar? "))

for i in range(0, casos):
    numero = int(input("Digite um número: "))

    if numero == 0:
        print("NULO")
    elif numero % 2 == 1:
        if numero < 0:
            print("IMPAR NEGATIVO")
        else:
            print("IMPAR POSITIVO")
    else:
        if numero < 0:
            print("PAR NEGATIVO")
        else:
            print("PAR POSITIVO")
